src/Utilities/FileSystem.cpp

```diff
@@ -153,21 +142,12 @@ bool create_directory(std::string dir, const double wait_time,
   }
 
   // Termination conditions
-  if (terminate_on_failure) {
-    if (number_of_failures >= num_tries) {
-      // LCOV_EXCL_START
-      ERROR("Unable to mkdir '" << dir << "'. Giving up after "
-                                << number_of_failures << " tries\n");
-      // LCOV_EXCL_STOP
-    }
-    if (got_stuck_in_infinite_loop) {
-      // LCOV_EXCL_START
-      ERROR("Got stuck in an infinite loop while trying to create '" << dir
-                                                                     << "'.\n");
-      // LCOV_EXCL_STOP
-    }
+  if (number_of_failures >= num_tries) {
```

Comment:
If we change the success case above to return then we will know getting here is an error and won't have to check.